File: scalang/src/main/scala/com/kapilkoju/autopos/kernel/api/CrudApi.scala
package com.kapilkoju.autopos.kernel.api

import com.kapilkoju.autopos.kernel.domain.AuditableEntity
import com.kapilkoju.autopos.kernel.service.AuditableRepository

abstract class CrudApi[T <: AuditableEntity](override protected val repo: AuditableRepository[T],
                                             override protected val entityName: String,
                                             override protected val baseUrl: String)
  extends Api[T](repo, entityName, baseUrl)
    with CreateApi[T]
    with GetApi[T]
    with UpdateApi[T]
    with DeleteApi[T]

File: src/main/java/com/kapilkoju/autopos/kernel/api/CrudApi.kt
package com.kapilkoju.autopos.kernel.api

import com.kapilkoju.autopos.kernel.domain.AuditableEntity
import com.kapilkoju.autopos.kernel.service.AuditableRepository

abstract class CrudApi<T : AuditableEntity>(override val repo: AuditableRepository<T>,
                                            override val entityName: String,
                                            override val baseUrl: String)
    : Api<T>, CreateApi<T>, GetApi<T>, UpdateApi<T>, DeleteApi<T>
